src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -200,7 +201,7 @@ private void processBatch(Request request, Context context, Promise<Response> oa
           return;
         }
 
-        if (!instances.isEmpty() && instances.get(0).getString(INSTANCE_ID_FIELD_NAME).equals(request.getNextRecordId())) {
+        if (!firstBatch && (CollectionUtils.isNotEmpty(instances) && !instances.get(0).getString(INSTANCE_ID_FIELD_NAME).equals(request.getNextRecordId()))) {
```

Comment:
We should respond with an error in the case when instance ids are NOT equal.

Reply:
Consider adding a comment in the code instead of here in the PR.